static/app/utils/url/isImageUrl.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Checks if a URL points to an image file based on common image extensions.
+ * The URL is trimmed and checked for extensions listed in the imageExtensionsPattern.
+ *
+ * @param url - The URL to check.
+ * @returns True if the URL ends with an image extension; otherwise, false.
+ */
+export function isImageUrl(url: string): boolean {
+  const imageExtensionsPattern = /\.(jpg|jpeg|png|gif|bmp|webp|svg)(\?.*)?$/i;
+  return imageExtensionsPattern.test(url.trim());
+}
```

Comment:
I think instead of checking extensions you could simply try to fetch the image data.
1. If it is not an image, the browser will abort fetching after receiving the `content-type` header. So there is not a big network overhead.
2. The extension in the URL is no proper indicator for the actual resource that is being served.

Reply:
Agree, there could be files that are images without extension